### PR TITLE
[ADD] website_event_question: new module to ask questions at registration

### PR DESCRIPTION
**Pull request by rim-odoo**
WIP...
- Task: https://www.odoo.com/web/#id=11596&view_type=form&model=project.task&action=333&active_id=248

### CODE REVIEW

File: addons/event/event.py [hunk 1]
```diff
@@ -303,6 +303,27 @@ def create(self, vals):
...
+        partner_id = registration.pop('partner_id', self.env.user.partner_id)
+        event_id = registration.pop('event_id', False)
+        data = {
+            'name': registration.get('name', partner_id.name),
+            'phone': registration.get('phone', partner_id.phone),
+            'email': registration.get('email', partner_id.email),
+            'partner_id': partner_id.id,
+            'event_id': event_id and event_id.id or False,
+        }
+        data.update({key: registration[key] for key in registration.keys() if key in self._all_columns})
```
Comment (tde-banana-odoo) [inline addons/event/event.py]: Note: I think we to use _fields now.

File: addons/event/event.py [hunk 2]
```diff
@@ -303,6 +303,27 @@ def create(self, vals):
...
         return registration
 
+    @api.model
+    def _prepare_attendee_values(self, registration):
+        """ Method preparing the values to create new attendees based on a
+        sale order line. It takes some registration data (dict-based) that are
+        optional values coming from an external input like a web page. This method
+        is meant to be inherited in various addons that sell events.
+
+        Kwargs can be used to update the method available parameter according to
```
Comment (tde-banana-odoo) [inline addons/event/event.py]: This old comment can be removed I think (don't know, I think I added this one before rewriting something else).

File: addons/event_sale/models/sale_order.py [hunk 3]
```diff
@@ -74,18 +75,9 @@ def _update_registrations(self, confirm=True, registration_data=None):
                 registration = {}
                 if registration_data:
                     registration = registration_data.pop()
-                # TDE CHECK: auto confirmation
```
Comment (tde-banana-odoo) [inline addons/event_sale/models/sale_order.py]: I would keep this comment, unless you are sur auto confirm is always correctly taken into account (still something to check before saas-6).

Comment (rim-odoo) [inline addons/event_sale/models/sale_order.py]: That was your code, but ok ;-)

File: addons/event/event.py (hunk 1 above)
Comment (rim-odoo) [inline addons/event/event.py]: So you are reviewing your own code :p

File: addons/website_event_questions/models/event.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    answer_ids = fields.Many2many('event.answer', 'event_registration_answer', string='Answers')
+
+
+class EventQuestion(models.Model):
+    _name = 'event.question'
+    _rec_name = 'title'
+    _order = 'sequence,id'
+
+    title = fields.Char(required=True)
```
Comment (tde-banana-odoo) [inline addons/website_event_questions/models/event.py]: Should'nt it be translatable ?

File: addons/website_event_questions/models/event.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    is_individual = fields.Boolean('Ask each attendee',
+                                   help="If True, this question will be asked for every attendee of a reservation. If "
+                                        "not it will be asked only once and its value propagated to every attendees.")
+
+
+class EventAnswer(models.Model):
+    _name = 'event.answer'
+    _order = 'sequence,id'
+
+    name = fields.Char('Answer', required=True)
```
Comment (tde-banana-odoo) [inline addons/website_event_questions/models/event.py]: Should'nt it be translatable ?

File: addons/website_event_questions/security/ir.model.access.csv [hunk 6]
```diff
@@ -0,0 +1,7 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+event_question_all,event.question.all,model_event_question,,1,0,0,0
+event_question_event_user,event.question.event.user,model_event_question,event.group_event_user,1,1,1,1
+event_answer_all,event.answer.all,model_event_answer,,1,0,0,0
+event_answer_event_user,event.answer.event.user,model_event_answer,event.group_event_user,1,1,1,1
+event_question_report_all,event.question.report.all,model_event_question_report,event.group_event_user,1,1,1,1
+event_registration_answer,access_event_registration_answer,model_event_registration_answer,,1,1,1,1
```
Comment (tde-banana-odoo) [inline addons/website_event_questions/security/ir.model.access.csv]: Generally we add the '_all' suffix for common (global) rules.

File: addons/website_event_questions/models/event.py (hunk 4 above)
Comment (rim-odoo) [inline addons/website_event_questions/models/event.py]: :+1:

File: addons/event/event.py (hunk 1 above)
Comment (tde-banana-odoo) [inline addons/event/event.py]: Better now than after merge :p !

Comment (rim-odoo) [inline addons/event/event.py]: Done, baby!

File: addons/event/event.py (hunk 2 above)
Comment (rim-odoo) [inline addons/event/event.py]: Done

File: addons/event_sale/models/sale_order.py (hunk 3 above)
Comment (rim-odoo) [inline addons/event_sale/models/sale_order.py]: Done

File: addons/website_event_questions/models/event.py (hunk 4 above)
Comment (rim-odoo) [inline addons/website_event_questions/models/event.py]: Done

File: addons/website_event_questions/models/event.py (hunk 5 above)
Comment (rim-odoo) [inline addons/website_event_questions/models/event.py]: Done

File: addons/website_event_questions/security/ir.model.access.csv (hunk 6 above)
Comment (rim-odoo) [inline addons/website_event_questions/security/ir.model.access.csv]: Done

**Comment by tde-banana-odoo** [comment]
:postal_horn:

**Pull request merged**